torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -106,16 +106,16 @@ inline Variable valueToTensor(
   at::AutoDispatchBelowADInplaceOrView guard; // TODO: remove
   at::tracer::impl::NoTracerDispatchMode tracer_guard;
   if (THPUtils_checkLong(value) || PyBool_Check(value)) {
-    return at::indexing::scalarToTensor(
-        Scalar(THPUtils_unpackLong(value)), options, device);
+    return at::lift_fresh(at::indexing::scalarToTensor(
+        Scalar(THPUtils_unpackLong(value)), options, device));
   }
   if (PyFloat_Check(value)) {
-    return at::indexing::scalarToTensor(
-        Scalar(THPUtils_unpackDouble(value)), options, device);
+    return at::lift_fresh(at::indexing::scalarToTensor(
+        Scalar(THPUtils_unpackDouble(value)), options, device));
   }
   if (PyComplex_Check(value)) {
-    return at::indexing::scalarToTensor(
-        Scalar(THPUtils_unpackComplexDouble(value)), options, device);
+    return at::lift_fresh(at::indexing::scalarToTensor(
```

Comment:
random side-car comment: `scalarToTensor` has a slow path that eventually calls `at::scalar_tensor`, and that op doesn't use `lift()` - I wonder if that will bite us anywhere else (basically, if there are any other user-facing API's that can cause us to hit `scalar_tensor` without going through `scalarToTensor`)